tests/integration/devfile/cmd_devfile_delete_test.go

```diff
@@ -31,187 +38,250 @@ var _ = Describe("odo devfile delete command tests", func() {
+				helper.CmdShouldPass("odo", "delete", "-f", "--all")
 
-			helper.CmdShouldPass("odo", "delete", "--project", commonVar.Project, "-f", "--all")
+				files := helper.ListFilesInDir(commonVar.Context)
+				Expect(files).To(Not(ContainElement(".odo")))
+				Expect(files).To(Not(ContainElement("devfile.yaml")))
+			})
 
-			commonVar.CliRunner.WaitAndCheckForExistence("deployments", commonVar.Project, 1)
+			Context("deleting a component from other component directory", func() {
+				var secondComp = helper.RandString(6)
+				var firstComp, firstDir, secondDir string
+
+				JustBeforeEach(func() {
+					// for the sake of verbosity
+					firstComp = componentName
+					firstDir = commonVar.Context
+					secondDir = createNewContext()
+					helper.Chdir(secondDir)
+				})
+				JustAfterEach(func() {
+					helper.Chdir(commonVar.Context)
+				})
+				When("the second component is created", func() {
+					JustBeforeEach(func() {
+						helper.CmdShouldPass("odo", "create", "nodejs", secondComp, "--project", commonVar.Project)
+					})
+					JustAfterEach(func() {})
+					When("the second component is pushed", func() {
+						JustBeforeEach(func() {
+							helper.CmdShouldPass("odo", "push", "--project", commonVar.Project)
+						})
+						JustAfterEach(func() {})
+						Context("delete with --context flag", func() {
+							It("should delete the context directory's component", func() {
+								output := helper.CmdShouldPass("odo", "delete", "--context", firstDir, "-f")
+								Expect(output).To(ContainSubstring(firstComp))
+								Expect(output).ToNot(ContainSubstring(secondComp))
+							})
+
+							It("should delete all the config files and component of the context directory with --all flag", func() {
+								output := helper.CmdShouldPass("odo", "delete", "--all", "--context", firstDir, "-f")
+								Expect(output).ToNot(ContainSubstring(secondComp))
+								Expect(output).To(ContainSubstring(firstComp))
+
+								files := helper.ListFilesInDir(firstDir)
+								Expect(files).To(Not(ContainElement(".odo")))
+								Expect(files).To(Not(ContainElement("devfile.yaml")))
+
+								files = helper.ListFilesInDir(secondDir)
+								Expect(files).To(ContainElement(".odo"))
+								Expect(files).To(ContainElement("devfile.yaml"))
+							})
+						})
+						// Marked as pending because it does not work at the moment. It takes the component in current directory into account while deleting.
+						XIt("should delete with the component name", func() {
+							output := helper.CmdShouldPass("odo", "delete", firstComp, "--app", "app", "--project", commonVar.Project, "-f")
+							Expect(output).ToNot(ContainSubstring(secondComp))
+							Expect(output).To(ContainSubstring(firstComp))
+						})
+					})
+				})
+			})
+		})
 
-			files := helper.ListFilesInDir(commonVar.Context)
-			Expect(files).To(Not(ContainElement(".odo")))
-			Expect(files).To(Not(ContainElement("devfile.yaml")))
+		Context("invalid delete commands", func() {
```

Comment:
This will run 5 independent tests. We could save some time by testing all the failures in the same test.

You can use `By` to document the different steps of the test.

Reply:
I did not think of that! `By` is nice! I will fix it.